lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -80,20 +75,107 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
 		return nil, err
 	}
 
+	subProtos, nextToken, err := loadSubProtosPage(req.PageSize, req.PageToken, networkID, apnsByName, apnResourcesByAPN)
+	if err != nil {
+		return nil, err
+	}
+
+	digest, noUpdates := s.getDigestAndNoUpdates(req, networkID, apnsByName, apnResourcesByAPN)
+	if noUpdates {
+		subProtos = []*lte_protos.SubscriberData{}
+	}
+
+	listRes := &lte_protos.ListSubscribersResponse{
+		Subscribers:   subProtos,
+		NextPageToken: nextToken,
+		Digest:        digest,
+		NoUpdates:     noUpdates,
+	}
+
+	return listRes, nil
+}
+
+func (s *subscriberdbServicer) getDigestAndNoUpdates(
```

Comment:
Nit: I don't like names with "and" in them. Just add the return values to a docstring for the method

```suggestion
func (s *subscriberdbServicer) getDigestInfo(
```